packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -137,7 +137,7 @@ class TimeInput extends React.Component {
     let hour = TimeUtil.splitHour(value);
     let meridiem;
 
-    if (props.variant === TimeUtil.FORMAT_12_HOUR) {
```

Comment:
We could reduce this call if we assigned the value:

```diff
+ const variant = TimeUtil.getVariantFromLocale(this.props)

 if (
   this.props.value === prevProps.value
+  && TimeUtil.getVariantFromLocale(this.props) === TimeUtil.getVariantFromLocale(prevProps)
-  && variant=== TimeUtil.getVariantFromLocale(prevProps)
 ) {
  return;
 }


   let hour = TimeUtil.splitHour(this.props.value);
   let { meridiem } = this.state;

- if (TimeUtil.getVariantFromLocale(this.props) === TimeUtil.FORMAT_12_HOUR) {
+ if (variant === TimeUtil.FORMAT_12_HOUR) {
```

Reply:
I think this comment was intended for the changes made in `componentDidUpdate`. Updated.